camel-openmrs-fhir/src/main/java/org/openmrs/eip/fhir/spring/OpenmrsFhirConfiguration.java

```diff
@@ -42,6 +43,23 @@ public boolean isOauthEnabled() {
 		return isOauthEnabled;
 	}
 	
+	@Qualifier("openmrsFhirClient")
+	@Bean
+	IGenericClient openmrsFhirClient() {
+		FhirContext ctx = FhirContext.forR4();
+		IGenericClient client = ctx.newRestfulGenericClient(fhirServerUrl);
+		
+		if (isOauthEnabled()) {
+			client.registerInterceptor(oauth2Interceptor);
+		} else {
+			if (fhirUsername != null && !fhirUsername.isBlank() && fhirPassword != null && !fhirPassword.isBlank()) {
```

Comment:
You can use StringUtils.isNotBlank from apache-commons or Spring which check for nulls, blanks, whitespace etc.

Reply:
Can't you combine this into **else if**? Instead of separate **else** and **if**

Reply:
It would be nice to have unit tests since you have some branching logic